TaskEngine/Tasks/CleanUpElasticIndexTask.cs

```diff
@@ -40,47 +39,49 @@ protected async override Task OnConsume(string example, TaskParameters taskParam
             registerTask(cleanup, "CleanUpElasticIndexTask"); // may throw AlreadyInProgress exception
             _logger.LogInformation("CleanUpElasticIndexTask Starting");
 
-            //using (var _context = CTDbContext.CreateDbContext())
-            //{
-            //    CaptionQueries captionQueries = new CaptionQueries(_context);
-
-            //    var all_transcriptions = await _context.Transcriptions.Where(t => t.Language == Languages.ENGLISH_AMERICAN).ToListAsync();
-            //    foreach (var transcription in all_transcriptions)
-            //    {
-            //        var all_captions = transcription.Captions;
+            var result = await _client.Indices.GetAsync(new GetIndexRequest(Indices.All));
+            var indices = result.Indices;
+            foreach (var index in indices)
+            {
+                string index_name = index.Key.ToString();
 
-            //        // each index has the unique name "index_string_unique", the current in use one has the alias "index_string_alias"
-            //        var index_string_base = transcription.Id + "_" + Languages.ENGLISH_AMERICAN.ToLower();
-            //        var index_string_unique = index_string_base + "_" + $"{DateTime.Now:yyyyMMddHHmmss}";
-            //        var index_string_alias = index_string_base + "_" + "primary";
+                // elastic search internal management index starts with '.'
+                if (index_name[0] == '.')
+                {
+                    continue;
```

Comment:
Can this code blow up; is it possible for a Key name to be null or empty? 
(In the former case then the type should be string?)

Reply:
No, it's not possible. ES would through an exception is index name is Null or Whitespace. Source: https://github.com/elastic/elasticsearch-net/blob/master/src/Nest/CommonAbstractions/Infer/IndexName/IndexNameResolver.cs#:~:text=%22Index%20name%20is%20null%20for,no%20default%20index%20is%20set.%20%22&text=%2B%20%22or%20set%20a%20default%20index,DefaultIndex().%22